src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilderCore.cs

```diff
@@ -123,7 +123,9 @@ internal static string GetAsyncStateMachineDescription(IAsyncStateMachine stateM
         }
 
         internal static Task? TryGetContinuationTask(Action continuation) =>
-            (continuation?.Target as ContinuationWrapper)?._innerTask;
+            (continuation?.Target is ContinuationWrapper wrapper) ?
+                wrapper._innerTask :            // A wrapped continuation, created by an awaiter
+                continuation?.Target as Task;   // The continuation targets a task directly, such as with AsyncStateMachineBox
```

Comment:
Looking at all the call sites, I don't believe it's possible for `continuation` to ever be null here, in which case the `?`s on `continuation?.` can be removed in both places.  But it's fine to not do so as part of this change.

Reply:
This is strange... by a applying this change, I'm now missing some operation started events. It doesn't make any sense, unless there are more issues with dotnet 5.

By the way, I have a suite of unit tests that I would like to execute these changes against to see what the impact is. vstest uses the installed version of core 3 to run the tests. Do you know of a way that I can patch my machine to make vstest run dotnet 5 with my changes instead?

Reply:
Never mind, I got it working. My local tests so far indicate that this will work as well.